src/ModValue.h

```diff
@@ -178,6 +178,48 @@ struct ModInput
 		}
 		return MIMT_Scalar;
 	}
+	// TODO: Make the field use needs_second to decide whether it needs to call
+	// GetElapsedTimeFromBeat at all. -Kyz
+	bool needs_beat()
+	{
+		switch(m_type)
+		{
+			case MIT_EvalBeat:
+			case MIT_MusicBeat:
+			case MIT_DistBeat:
+			case MIT_StartDistBeat:
+			case MIT_EndDistBeat:
+				return true;
+			default:
+				return false;
+		}
+	}
+	bool needs_second()
+	{
+		switch(m_type)
+		{
+			case MIT_EvalSecond:
+			case MIT_MusicSecond:
+			case MIT_DistSecond:
+			case MIT_StartDistSecond:
+			case MIT_EndDistSecond:
+				return true;
+			default:
+				return false;
+		}
+	}
+	bool needs_y_offset()
+	{
+		switch(m_type)
```

Comment:
I use the switch statements to make it easier to change in the future.  If someone invents a new type of y offset based input field, they just have to add it to the switch statement.